src/lightning/fabric/accelerators/tpu.py

```diff
@@ -143,40 +149,38 @@ def _parse_tpu_devices(devices: Optional[Union[int, str, List[int]]]) -> Optiona
             indicates the specific TPU core to use.
 
     Returns:
-        A list of tpu_cores to be used or ``None`` if no TPU cores were requested
-
-    Raises:
-        TypeError:
-            If TPU devices aren't 1, 8 or [<1-8>]
+        A list of tpu cores to be used.
     """
     _check_data_type(devices)
-
     if isinstance(devices, str):
-        devices = _parse_tpu_devices_str(devices.strip())
-
-    if not _tpu_devices_valid(devices):
-        raise TypeError("`devices` can only be 1, 8 or [<1-8>] for TPUs.")
```

Comment:
I think the device index was off by one when passing `devices=[n]`. Docs and this ask for 1-based index, but `torch.device` uses 0-based